lib/manageiq/appliance_console/saml_authentication.rb

```diff
@@ -0,0 +1,212 @@
+                            :params => ["https://#{host}", "https://#{host}/saml2"])
+        end
+        rename_mellon_configfiles
+        fetch_idp_metadata
+        configure_auth_settings_saml
+        restart_httpd
+        true
+      rescue AwesomeSpawn::CommandResultError => e
+        log_command_error(e)
+        say("Failed to Configure SAML Authentication - #{e}")
+        false
+      rescue => e
+        say("Failed to Configure SAML Authentication - #{e}")
+        false
+      end
+
+      def unconfigure
+        raise "Appliance is not currently configured for SAML" unless configured?
+
+        say("Unconfiguring SAML Authentication ...")
+        remove_apache_saml_configfiles
+        configure_auth_settings_database
+        restart_httpd
+        true
+      rescue AwesomeSpawn::CommandResultError => e
+        log_command_error(e)
+        say("Failed to Unconfigure SAML Authentication - #{e}")
+        false
+      rescue => e
+        say("Failed to Unconfigure SAML Authentication - #{e}")
+        false
+      end
+
+      private
+
+      # Apache SAML Configuration
+
+      def rename_mellon_configfiles
+        debug_msg("Renaming mellon config files ...")
+        Dir.chdir(SAML2_CONFIG_DIRECTORY) do
+          Dir.glob("https_*.*") do |mellon_file|
+            saml2_file = nil
+            case mellon_file
+            when /^https_.*\.key$/  then saml2_file = "miqsp-key.key"
+            when /^https_.*\.cert$/ then saml2_file = "miqsp-cert.cert"
+            when /^https_.*\.xml$/  then saml2_file = "miqsp-metadata.xml"
+            end
+            if saml2_file
+              debug_msg("Renaming #{mellon_file} to #{saml2_file}")
+              File.rename(mellon_file, saml2_file)
+            end
+          end
+        end
+      end
+
+      def fetch_idp_metadata
+        idp_metadata = options[:saml_idp_metadata]
+        if path_is_file(idp_metadata) && idp_metadata != IDP_METADATA_FILE
+          debug_msg("Copying IDP metadata file #{idp_metadata} to #{IDP_METADATA_FILE} ...")
+          FileUtils.cp(idp_metadata, IDP_METADATA_FILE)
+        elsif path_is_url(idp_metadata)
+          debug_msg("Downloading IDP metadata file from #{idp_metadata}")
+          download_network_file(idp_metadata, IDP_METADATA_FILE)
+        end
+      end
+
+      def copy_apache_saml_configfiles
+        debug_msg("Copying Apache SAML Config files ...")
+        copy_template(HTTPD_CONFIG_DIRECTORY, "manageiq-remote-user.conf")
+        copy_template(HTTPD_CONFIG_DIRECTORY, "manageiq-external-auth-saml.conf")
+      end
+
+      def remove_apache_saml_configfiles
+        debug_msg("Removing Apache SAML Config files ...")
+        remove_file(HTTPD_CONFIG_DIRECTORY, "manageiq-remote-user.conf")
+        remove_file(HTTPD_CONFIG_DIRECTORY, "manageiq-external-auth-saml.conf")
+      end
+
+      def configured?
+        path_join(HTTPD_CONFIG_DIRECTORY, "manageiq-external-auth-saml.conf").exist?
```

Comment:
Can this just be `HTTPD_CONFIG_DIRECTORY.join("manageiq-external-auth-saml.conf").exist?`